src/Frontend/Driver.hs

```diff
@@ -261,32 +268,48 @@ compile path = do
   case ok of
     Nothing -> pure (Nothing, errs)
     Just () -> do
-      (_, _, lowered) <- lowerIt True path mempty
-      pure (Just (concat lowered), errs)
+      mod <- findMod path
+      deps <- getDependencyGraph mod
+      (_, _, lowered) <- lowerIts True deps mempty
+      pure (Just (concat (Seq.reverse lowered)), errs)
+  where
+    getDependencyEdgeList mod = do
+      let deps_list = toList (mod ^. dependencies)
+
+      edges <- concat <$> for deps_list \dep -> do
+        mod <- findMod dep
+        getDependencyEdgeList mod
+      pure ((modLocation mod, modLocation mod, deps_list):edges)
+
+    getDependencyGraph mod = do
+      edges <- getDependencyEdgeList mod
+      let (graph, get_vertex, _) = graphFromEdges edges
+          vertices = topSort graph
+          files = map (view _1 . get_vertex) vertices
+      pure (nub (reverse files))
 
 lowerIt :: (MonadNamey m, MonadState Driver m)
         => Bool -> FilePath
         -> (Set.Set FilePath, LowerState, Seq.Seq [Stmt CoVar])
         -> m (Set.Set FilePath, LowerState, Seq.Seq [Stmt CoVar])
 lowerIt all path (visited, lEnv, stmts)
-  | path `Set.member` visited = pure (visited, lEnv, stmts)
+  -- | path `Set.member` visited = pure (visited, lEnv, stmts)
```

Comment:
Can we remove the visited flag entirely then? I think my original code was technically fine, it should have just been recursing and wasn't for some reason :/.